src/code/z_skelanime.c

```diff
@@ -987,21 +994,25 @@ AnimationEntry* AnimationContext_AddEntry(AnimationContext* animationCtx, Animat
     return entry;
 }
 
+#define LINK_ANIMETION_OFFSET(addr, offset) \
+    (SEGMENT_ROM_START(link_animetion) + ((uintptr_t)addr & 0xFFFFFF) + ((u32)offset))
+
 /**
- * Requests loading frame data from the Link animation into frameTable
+ * Requests loading frame data from the Player animation into frameTable
  */
-void AnimationContext_SetLoadFrame(PlayState* play, LinkAnimationHeader* animation, s32 frame, s32 limbCount,
+void AnimationContext_SetLoadFrame(PlayState* play, PlayerAnimationHeader* animation, s32 frame, s32 limbCount,
                                    Vec3s* frameTable) {
     AnimationEntry* entry = AnimationContext_AddEntry(&play->animationCtx, ANIMATION_LINKANIMETION);
 
     if (entry != NULL) {
-        LinkAnimationHeader* linkAnimHeader = Lib_SegmentedToVirtual(animation);
-        uintptr_t ram = frameTable;
+        PlayerAnimationHeader* playerAnimHeader = Lib_SegmentedToVirtual(animation);
+        void* ram = (void*)frameTable;
 
-        osCreateMesgQueue(&entry->data.load.msgQueue, &entry->data.load.msg, 1);
-        DmaMgr_SendRequestImpl(&entry->data.load.req, ram,
-                               LINK_ANIMETION_OFFSET(linkAnimHeader->segment, (sizeof(Vec3s) * limbCount + 2) * frame),
-                               sizeof(Vec3s) * limbCount + 2, 0, &entry->data.load.msgQueue, NULL);
+        osCreateMesgQueue(&entry->data.load.msgQueue, entry->data.load.msg, ARRAY_COUNT(entry->data.load.msg));
+        DmaMgr_SendRequestImpl(
+            &entry->data.load.req, ram,
+            LINK_ANIMETION_OFFSET(playerAnimHeader->segment, (sizeof(Vec3s) * limbCount + sizeof(s16)) * frame),
```

Comment:
Yes, `LINK_ANIMETION_OFFSET` is named like that because it is accessing the `link_animetion` segment and I didn't want to rename it in this PR.
And yeah, the inconsistency is annoying. Would be better if I added a comment explaining it?

Reply:
Does `playerAnimHeader->segment` ever point to something else? Woudl `playerAnimHeader->linkAnimSegment` be too verbose?

Reply:
I think `playerAnimHeader->segment` always points to that segment, so yeah I think naming that member that way should be fine